src/services/env/Env.ts

```diff
@@ -1,48 +1,55 @@
 import { PATH_CONFIG_DEFAULT } from '@/pathConfigDefault'
 import { PathConfig } from '@/types/index'
 
-export type EnvVars = {
-  KUMA_NAME: string
+export type EnvArgs = {
+  KUMA_PRODUCT_NAME: string
+  KUMA_FEEDBACK_URL: string
+  KUMA_CHAT_URL: string
+  KUMA_INSTALL_URL: string
+  KUMA_DOCS_URL: string
+}
+type EnvProps = {
   KUMA_VERSION: string
   KUMA_BASE_PATH: string
   KUMA_API_URL: string
-  KUMA_DOCS_URL: string
   KUMA_UTM_QUERY_PARAMS: string
-  KUMA_FEEDBACK_URL: string
-  KUMA_CHAT_URL: string
-  KUMA_INSTALL_URL: string
 }
+export type EnvVars = EnvArgs & EnvProps
+
+type EnvInternal = EnvArgs & Partial<EnvProps>
 export default class Env {
   env: EnvVars
-  constructor(env: EnvVars) {
-    this.env = env
+  constructor(envArgs: EnvArgs) {
+    let _env: EnvInternal = envArgs
+    const env = (str: keyof EnvInternal, d: string = '') => this.var(str, this.var(str, _env?.[str] ?? d))
+
+    const config = readPathConfigFromDom()
+    const version = semver(env('KUMA_VERSION', config.version))
+
+    _env = {
+      ..._env,
+      ...{
+        KUMA_UTM_QUERY_PARAMS: `utm_source=${env('KUMA_PRODUCT_NAME')}&utm_medium=${env('KUMA_PRODUCT_NAME')}-GUI`,
+      },
```

Comment:
**Suggestion**: Please replace the unnecessary destructuring with setting properties directly.